--- initrd.scripts.orig	2022-11-19 19:52:04.944110136 +0100
+++ initrd.scripts	2022-11-19 19:58:02.304271001 +0100
@@ -2067,9 +2067,32 @@
 				good_msg "Using the following cryptsetup options for ${LUKS_NAME}: ${cryptsetup_options}" ${CRYPT_SILENT}
 			fi
 
-			# At this point, {header,key}file or not, we're ready!
-			crypt_filter "${gpg_cmd}cryptsetup ${cryptsetup_options} luksOpen ${LUKS_DEVICE} ${LUKS_NAME}" "${ASK_PASS}"
-			crypt_filter_ret=$?
+                        # nitrokey
+                        NITROLUKS_BIN="/bin/nitro_luks"
+                        local nitropass="$(mktemp -t "nitropass.XXXXXX")"
+                        $NITROLUKS_BIN > $nitropass
+                        local retnitro="${?}"
+
+                        if [ "${retnitro}" = "0" ]; then
+                                echo "Nitrokey pro opened"
+                                crypt_filter "cat ${nitropass} | ${gpg_cmd}cryptsetup ${cryptsetup_options} luksOpen ${LUKS_DEVICE} ${LUKS_NAME}"
+                                crypt_filter_ret=$?
+                                if [ "${crypt_filter_ret}" = "0" ]; then
+                                        rm -f $nitropass
+                                else
+                                        echo "error on nitrokey cryptsetup"
+                                        rm -f $nitropass
+			                # At this point, {header,key}file or not, we're ready!
+                                        crypt_filter "${gpg_cmd}cryptsetup ${cryptsetup_options} luksOpen ${LUKS_DEVICE} ${LUKS_NAME}" "${ASK_PASS}"
+                                        crypt_filter_ret=$?
+                                fi
+                        else
+                                echo "error on nitrokey"
+                                rm -f $nitropass
+			         # At this point, {header,key}file or not, we're ready!
+                                crypt_filter "${gpg_cmd}cryptsetup ${cryptsetup_options} luksOpen ${LUKS_DEVICE} ${LUKS_NAME}" "${ASK_PASS}"
+                                crypt_filter_ret=$?
+                        fi # end if [ "${retnitro}" = "0" ]
 
 			[ -e /dev/tty.org ] \
 				&& run rm -f /dev/tty \
